### PR TITLE
Add validateDomainName to ValidatesAttributes

### PR DESCRIPTION
**Pull request by a3dho3yn**
There were no validation rules for [domain names](https://en.wikipedia.org/wiki/Domain_name) in the framework. This simple method validates an input against [rfc1035](https://tools.ietf.org/html/rfc1035).

### CODE REVIEW

File: src/Illuminate/Validation/Concerns/ValidatesAttributes.php [hunk 1]
```diff
@@ -1741,6 +1741,38 @@ public function validateUrl($attribute, $value)
         return preg_match($pattern, $value) > 0;
     }
 
+    /**
+     * Validate that an attribute is a valid URL.
```
Comment (browner12) [inline src/Illuminate/Validation/Concerns/ValidatesAttributes.php]: we are validating a domain, not a URL, so update this comment appropriately.

maybe even add the RFC reference in here.  "according to RFC..."

File: src/Illuminate/Validation/Concerns/ValidatesAttributes.php [hunk 2]
```diff
@@ -1741,6 +1741,38 @@ public function validateUrl($attribute, $value)
...
+        }
+
+        $labels = explode(".", $value);
+        if (count($labels) < 2) {
+            return false;
+        }
+
+        // ToDo get TLDs from https://data.iana.org/TLD/tlds-alpha-by-domain.txt and compare first label with them
+        foreach ($labels as $l) {
+            if (! preg_match("/^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9]|)$/D", $l)) {
```
Comment (browner12) [inline src/Illuminate/Validation/Concerns/ValidatesAttributes.php]: don't use abbreviated variable names.  `$l` --> `$label`

**Comment by browner12** [comment]
I think the easier answer here might be using `filter_var`, which validates specifically against RFC 1035.

```php
return filter_var($value, FILTER_VALIDATE_DOMAIN) !== false;
```

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by a3dho3yn** [comment]
Without `FILTER_FLAG_HOSTNAME` flag,`filter_var` doesn't work as expected. It recognizes `com`, `-1.com`, `1-.com` and `1_1.com` as valid domains!
So, we should write:
```php
return filter_var($value, FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME) !== false;
```

But, as the [ToDo](https://github.com/laravel/framework/pull/31129/files#diff-bfe2e79d87a34030e1cd049713a78cdaR1766) shows, it will be more robust to check the TLD against the list of TLDs that IANA provided.

**Pull request closed without merging**